File: sdk-python/examples/chat_completion_example.py
#!/usr/bin/env python3
"""
Chat Completion Example

This example demonstrates how to use the Pearl SDK to send chat completion requests.

To run this example:
1. Install the SDK: pip install pearl-sdk
2. Set your API key in the API_KEY variable below
3. Run: python examples/chat_completion_example.py
"""

import uuid
from pearl_sdk import PearlClient, ChatMessage, RetryPolicyConfig, ConversationModes

# --- IMPORTANT: Configure your API Key ---
# Replace 'YOUR_PEARL_API_KEY' with your actual Pearl API Key.
API_KEY = 'YOUR_PEARL_API_KEY'


def run_chat_completion_example():
    """Run the chat completion example."""
    if API_KEY == 'YOUR_PEARL_API_KEY':
        print("WARNING: Please replace 'YOUR_PEARL_API_KEY' with your actual API key to make live API calls.")
        print("Exiting example as API Key is a placeholder.")
        return

    session_id = str(uuid.uuid4())
    # Initialize client with comprehensive configuration for demonstration
    client = PearlClient(
        api_key=API_KEY,
        # base_url='https://api.pearl.com/api/v1',  # Uncomment and change if using a custom base URL
        timeout=60,  # 60 seconds timeout
        retry_policy=RetryPolicyConfig(
            enabled=True,
            max_retries=50,
            retry_delay_ms=100,
            max_retry_delay_ms=60000        )
    )

    try:
        print("\n--- Starting Chat Completion Example ---")

        messages = [
            ChatMessage(
                role="user",
                content="Explain large language models in simple terms."  # The user's message
            )
        ]

        print("Sending chat completion request...")
        chat_response = client.chat.send_completion(
            messages=messages,
            session_id=session_id,
            mode=ConversationModes.PEARL_AI  # optional, defaults to PEARL_AI
        )

        print("\n--- Chat Completion API Response ---")
        print("Assistant's message:", chat_response.choices[0].message.content)

    except Exception as error:
        print("\n--- Chat Completion API Error ---")
        print(f"An error occurred: {error}")
        
        # Additional error handling for requests exceptions
        if hasattr(error, 'response'):
            print(f"Status code: {error.response.status_code}")
            print(f"Response: {error.response.text}")

    finally:
        print("\n--- End of Chat Completion Example ---")


if __name__ == "__main__":
    run_chat_completion_example()

File: sdk-typescript/examples/chat-completion-example.ts
// To run this example:
// 1. Ensure you have the SDK installed: npm install pearl-sdk
// 2. Transpile and run:
//    npx ts-node examples/chat-completion-example.ts

import { PearlClient, CONVERSATION_MODES, ChatMessage } from 'pearl-sdk';
import { v4 as uuidv4 } from 'uuid'; // Ensure uuid is installed: npm install uuid
import axios from 'axios';

// --- IMPORTANT: Configure your API Key ---
// Replace 'YOUR_PEARL_API_KEY' with your actual Pearl API Key.
// For demonstration purposes, you can use a placeholder, but a real key is needed for live calls.
const API_KEY = 'YOUR_PEARL_API_KEY';

async function runChatCompletionExample() {
  if (API_KEY === 'YOUR_PEARL_API_KEY') {
    console.warn("WARNING: Please replace 'YOUR_PEARL_API_KEY' with your actual API key to make live API calls.");
    console.log("Exiting example as API Key is a placeholder.");
    return;
  }
  const sessionId = uuidv4();
  // Initialize client with comprehensive configuration for demonstration
  const client = new PearlClient(API_KEY, {
    timeout: 60000, // 60 seconds timeout
    retryPolicy: {
      enabled: true,
      maxRetries: 50,
      retryDelayMs: 100,
      maxRetryDelayMs: 60000
    }
  });

  try {
    console.log("\n--- Starting Chat Completion Example ---");

    const messages = [
      {
        role: "user",
        content: "Explain large language models in simple terms." // The user's message
      }
    ] as ChatMessage[];
    console.log("Sending chat completion request...");
    const chatResponse = await client.chat.sendCompletion(
      messages,
      sessionId,
      CONVERSATION_MODES.PEARL_AI // mode (optional, defaults to PEARL_AI)
    );

    console.log("\n--- Chat Completion API Response ---");
    console.log("Assistant's message:", chatResponse.choices[0].message.content);

  } catch (error: any) {
    console.error("\n--- Chat Completion API Error ---");
    // Axios errors are handled by PearlClient's interceptors, but this catch
    // demonstrates how to handle final rejections from the SDK call.
    if (axios.isAxiosError(error)) {
      console.error("Error message:", error.message);
      if (error.response) {
        console.error("Status:", error.response.status);
        console.error("Data:", JSON.stringify(error.response.data, null, 2));
      } else if (error.request) {
        console.error("No response received from server.");
      } else {
        console.error("Error setting up request.");
      }
    } else {
      console.error("An unexpected error occurred:", error);
    }
  } finally {
    console.log("\n--- End of Chat Completion Example ---");
  }
}

// Run the example
runChatCompletionExample();